openedx/features/course_duration_limits/access.py

```diff
@@ -0,0 +1,59 @@
+# -*- coding: utf-8 -*-
+"""
+Contains code related to computing content gating course duration limits
+and course access based on these limits.
+"""
+from datetime import timedelta
+
+from django.apps import apps
+from django.utils import timezone
+from django.utils.translation import ugettext as _
+
+
+from lms.djangoapps.courseware.access_response import AccessError
+from lms.djangoapps.courseware.access_utils import ACCESS_GRANTED
+
+
+class AuditExpiredError(AccessError):
+    """
+    Access denied because the user's audit timespan has expired
+    """
+    def __init__(self, user, course, expiration_date):
+        error_code = "audit_expired"
+        developer_message = "User {} had access to {} until {}".format(user, course, expiration_date)
+        # TODO: Translate the expiration_date
+        user_message = _("Course access expired on ") + expiration_date.strftime("%B %d, %Y")
```

Comment:
I haven't figured out how to localize this date in edX platform and test it locally. (I seem to only have English available as a language in my local devstack)